h2/src/main/org/h2/value/Transfer.java

```diff
@@ -494,6 +494,20 @@ public void writeValue(Value v) throws IOException {
             }
             break;
         }
+        case Value.ENUM: {
+            ValueEnum ve = (ValueEnum) v;
+            String[] enumerators = ve.getEnumerators();
+
+            writeInt(enumerators.length);
+
+            for (String member : enumerators) {
+                writeString(member);
+            }
+
+            writeInt(ve.getInt());
+
+            break;
+        }
```

Comment:
Yes, that would work fine.

The other option would be to have two ValueEnum classes - one for the server side, and one for the client side. The client side one would only need to know the integer id and the string value, because it would only be used to pass data through the JDBC interface. The server side one would need the full set of enumerators,